tests/test_distribution.py

```diff
@@ -0,0 +1,216 @@
+from pythonforandroid.distribution import Distribution
+from pythonforandroid.recipe import Recipe
+from pythonforandroid.util import BuildInterruptingException
+from pythonforandroid.build import Context
+
+dist_info_data = {
+    "dist_name": None,
+    "bootstrap": "sdl2",
+    "archs": ["armeabi", "armeabi-v7a", "x86", "x86_64", "arm64-v8a"],
+    "ndk_api": 21,
+    "use_setup_py": False,
+    "recipes": ["hostpython3", "python3", "sdl2", "kivy", "requests"],
+    "hostpython": "/some/fake/hostpython3",
+    "python_version": "3.7",
+}
+
+
+class TestDistribution(unittest.TestCase):
+    def setUp(self):
+        self.ctx = Context()
+        self.ctx.ndk_api = 21
+        self.ctx.android_api = 27
+        self.ctx._sdk_dir = "/opt/android/android-sdk"
+        self.ctx._ndk_dir = "/opt/android/android-ndk"
+        self.ctx.setup_dirs(os.getcwd())
+        self.ctx.recipe_build_order = [
+            "hostpython3",
+            "python3",
+            "sdl2",
+            "kivy",
+        ]
+
+    def setUp_distribution_with_bootstrap(self, bs, **kwargs):
+        # extend the setUp by configuring a distribution, because some test
+        # needs a distribution to be set to be properly tested
+        self.ctx.bootstrap = bs
+        self.ctx.bootstrap.distribution = Distribution.get_distribution(
+            self.ctx,
+            name=kwargs.pop("name", "test_prj"),
+            recipes=kwargs.pop("recipes", ["python3", "kivy"]),
+            **kwargs
+        )
+
+    def tearDown(self):
+        self.ctx.bootstrap = None
+
+    def test_properties(self):
+        self.setUp_distribution_with_bootstrap(
+            Bootstrap().get_bootstrap("sdl2", self.ctx)
+        )
+        distribution = self.ctx.bootstrap.distribution
+        self.assertEqual(self.ctx, distribution.ctx)
+        expected_repr = (
+            "<Distribution: name test_prj with recipes (python3, kivy)>"
+        )
+        self.assertEqual(distribution.__str__(), expected_repr)
+        self.assertEqual(distribution.__repr__(), expected_repr)
+
+    @mock.patch("pythonforandroid.distribution.exists")
+    def test_folder_exist(self, mock_exists):
+
+        self.setUp_distribution_with_bootstrap(
+            Bootstrap().get_bootstrap("sdl2", self.ctx)
+        )
+        self.ctx.bootstrap.distribution.folder_exists()
+        mock_exists.assert_called_with(
+            self.ctx.bootstrap.distribution.dist_dir
+        )
+
+    @mock.patch("pythonforandroid.distribution.rmtree")
+    def test_delete(self, mock_rmtree):
+
+        self.setUp_distribution_with_bootstrap(
+            Bootstrap().get_bootstrap("sdl2", self.ctx)
+        )
+        self.ctx.bootstrap.distribution.delete()
+        mock_rmtree.assert_called_with(
+            self.ctx.bootstrap.distribution.dist_dir
+        )
+
+    @mock.patch("pythonforandroid.distribution.exists")
+    def test_get_distribution_no_name(self, mock_exists):
+
+        mock_exists.return_value = False
```

Comment:
Minor: not sure we need to mock `exists()` here since `folders` from `get_distributions()` is empty so we won't get into the loop right?